syntax = "proto3";

option go_package = "/protobufs";

service PestClusteringService {
    rpc ClusterizeSpread (ClusterizeSpreadRequest) returns (ClusterizeSpreadResponse);
}

message DeltaData {
    string forest = 1;
    string plot = 2;
    int32 delta_min = 3;
    int32 delta_max = 4;
    int32 delta = 5;
    string start_date = 6;
    string end_date = 7;
    int32 x = 8;
    int32 y = 9;
    double ndre = 10;
    double ndmi = 11;
    double psri = 12;
    double ndvi = 13;
    double ndre_derivative = 14;
    double ndmi_derivative = 15;
    double psri_derivative = 16;
    double ndvi_derivative = 17;
    string label = 18;
    double latitude = 19;
    double longitude = 20;
}

message PestSpreadSample {
    DeltaData data = 1;
    int32 cluster = 2;
}

message ClusterizeSpreadRequest {
    repeated DeltaData delta_data = 1;
}

message ClusterizeSpreadResponse {
    repeated PestSpreadSample pest_spread_samples = 1;
} 